hrms/overrides/expense_claim.py

```diff
@@ -0,0 +1,359 @@
+        if not user_has_role(frappe.session.user, RoleEnum.FINANCE_ROLE.value):
+            frappe.throw(_("Only Admin approve the document"))
+        expense_claim_doc.docstatus = 1
+        expense_claim_doc.status = "Pending Approval By HR"
+        expense_claim_doc.db_update()
+    elif expense_claim_doc.status == "Pending Approval By HR":
+        if not user_has_role(frappe.session.user, RoleEnum.HR_ROLE.value):
+            frappe.throw(_("Only user having HR User can approve the document"))
+        if expense_claim_doc.grand_total == 0:
+            expense_claim_doc.status = "Paid"
+        else:
+            expense_claim_doc.status = "Pending Payment"
+        expense_claim_doc.db_update()
+
+
+@frappe.whitelist()
+def get_total_pending_amount(doc):
+    import json
+
+    if isinstance(doc, str):
+        doc = frappe._dict(json.loads(doc))
+
+    expense_claims = frappe.db.get_list(
+        "Expense Claim",
+        filters={"status": "Pending Payment", "employee_name": doc.employee_name},
+        fields=["grand_total"],
+        pluck="grand_total",
+    )
+    total_amount = sum(expense_claims)
+    return total_amount
+
+
+@frappe.whitelist()
+def get_mode_of_payments():
+    return frappe.db.get_list("Mode of Payment", pluck="name")
+
+
+@frappe.whitelist()
+def create_payment_entry(doc, values):
+    import json
+
+    if isinstance(doc, str):
+        doc = frappe._dict(json.loads(doc))
+    if isinstance(values, str):
+        payment_values = frappe._dict(json.loads(values))
+
+    party_details = get_party_details(
+        doc.company, "Employee", doc.employee, frappe.utils.nowdate()
+    )
+
+    bank_cash_account = get_bank_cash_account(payment_values.mode_of_payment, doc.company).get(
+        "account"
+    )
+    bank_account_currency = get_account_details(bank_cash_account, frappe.utils.nowdate()).get(
+        "account_currency"
+    )
+
+    expense_claims_list = frappe.db.get_list(
+        "Expense Claim",
+        filters={"status": "Pending Payment", "employee_name": doc.employee_name},
+        pluck="name",
+    )
+    reference_list = []
+    for name in expense_claims_list:
+        expense_claim_doc = frappe.get_doc("Expense Claim", name)
+        remaining_amount = (
+            expense_claim_doc.grand_total - expense_claim_doc.total_amount_reimbursed
+        )
+        reference_entry = {
+            "reference_doctype": "Expense Claim",
+            "reference_name": expense_claim_doc.name,
+            "total_amount": expense_claim_doc.grand_total,
+            "outstanding_amount": remaining_amount,
+            "allocated_amount": remaining_amount,
+            "parentfield": "references",
+            "parenttype": "Payment Entry",
+            "docstatus": 1,
+            "doctype": "Payment Entry Reference",
+        }
+        reference_list.append(reference_entry)
+
+    if doc.company == "One Suite Technologies Private Limited":
+        paid_to = "Reimbursement Payable - Suite42"
```

Comment:
fetch from one of the expense claims
validate all the expense claims have the same payable account

Reply:
Done.